admin.js

```diff
@@ -13,6 +13,35 @@ jQuery(function () {
         jQuery('.match input[id^=' + this.id.replace(/:/g, '\\:') + ']').prop('checked', this.checked);
     });
 
+    // When all single matches of a file have been checked, mark the appropriate "check all" box as checked, too.
+    jQuery('.check-single-match').click(function(){
+        var pageIdEscaped = this.id.substr(0, this.id.indexOf('#')).replace(/:/g, '\\:');
+        var checkedAll = true;
+        jQuery('.match input[id^=' + pageIdEscaped + ']').each(function(i, el) {
+            checkedAll = checkedAll && jQuery(el).prop('checked');
+            return checkedAll;
+        });
```

Comment:
Oh, I did not even know that exists in jQuery. Never really worked much with jQuery. I only really know more modern ES5 or ES6, which has already so many features of jQuery included as standard functions.
On a sidenote, do you think it might perhaps be better to use vanilla JS for certain jQuery features that have been incorporated into the standard at the latest in ES5? I think browsers that do not know at least ES5 JavaScript are rarely found anymore these days.

Reply:
It's not in jQuery, which was actually a [big surprise](https://bugs.jquery.com/ticket/1886) for me. As for pure JS, you might have noticed that I removed jQuery wrapper around el in example below. Hint, hint ;)